bowler/tool.py

```diff
@@ -281,14 +283,19 @@ def process_hunks(self, filename: Filename, hunks: List[Hunk]) -> None:
                         raise ValueError("unknown response")
 
             if result == "y" or self.write:
-                patch = ("\n".join(hunk) + "\n").encode("utf-8")
-                args = ["patch", "-u", filename]
-                self.log_debug(f"running {args}")
-                try:
-                    sh.patch("-u", filename, _in=patch)  # type: ignore
-                except sh.ErrorReturnCode:
-                    log.exception("failed to apply patch hunk")
-                    return  # TODO: better response?
+                need_patch = True
+                accepted_hunks += "\n".join(hunk[2:]) + "\n"
+
+        if need_patch:
+            args = ["patch", "-u", filename]
+            self.log_debug(f"running {args}")
+            try:
+                sh.patch(
+                    "-u", filename, _in=accepted_hunks.encode("utf-8")
+                )  # type: ignore
+            except sh.ErrorReturnCode:
+                log.exception("failed to apply patch hunk")
+                return  # TODO: better response?
```

Comment:
Agreed, a better response would be great. A good way to find out what response would be helpful is to make a test that exercises this path :-)